Appropriately Set gRPC Client Max Send and Receive Message Length

js8080 opened the pull request:
<!--
Please delete this comment before posting.

We appreciate your contribution to the Jaeger project! 👋🎉

Before creating a pull request, please make sure:
- Your PR is solving one problem
- You have read the guide for contributing
  - See https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING.md
- You signed all your commits (otherwise we won't be able to merge the PR)
  - See https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md#certificate-of-origin---sign-your-work
- You added unit tests for the new functionality
- You mention in the PR description which issue it is addressing, e.g. "Resolves #123"
-->

## Which problem is this PR solving?
#225 

## Short description of the changes
Even though the GrpcSender allowed specifying a maxPacketSize
parameter if that value was over 4MB, Spans over that 4MB limit
would not be recorded because the core gRPC client library has a
default max message length of 4MB. Now, the GrpcSender will set
the appropriate ChannelOptions to set the MaxSendMessageLength and
MaxReceiveMessageLength based on the given maxPacketSize.

Signed-off-by: Justin Stauffer <justin@epic.com>

Falco20019 commented:
Thanks for your report and the PR. I will give it a look tomorrow (UTC+1 here). Can you please apply the DCO fix like described here: https://github.com/jaegertracing/jaeger-client-csharp/runs/3344037372

js8080 commented:
> Thanks for your report and the PR. Can you please apply the DCO fix like described here: https://github.com/jaegertracing/jaeger-client-csharp/runs/3344037372
> 
> Just make sure that the mail used in the sign-off has to match the one used as author of the commit set in your got config. You can use a local got config and set the mail accordingly if you want to use your gmail instead of epic mail.
> 
> Besides that, the PR looks good to me and I will merge it once fixed.

I fixed the commit sign-off email. Also updated it in the PR description for consistency.
